modules/standard/Sys.chpl

```diff
@@ -208,22 +227,125 @@ module Sys {
   extern const TCP_SYNCNT:c_int;
   extern const TCP_WINDOW_CLAMP:c_int;
 
+  // socket address sizes
+  extern const INET_ADDRSTRLEN:c_int;
+  extern const INET6_ADDRSTRLEN:c_int;
+
+  // standard ipv4 addresses
+  extern const INADDR_ANY:sys_in_addr_t;
+  extern const INADDR_BROADCAST:sys_in_addr_t;
+  extern const INADDR_LOOPBACK:sys_in_addr_t;
+
+  // standard ipv6 addresses
+  extern const in6addr_any:sys_in6_addr_t;
+  extern const in6addr_loopback:sys_in6_addr_t;
+
   // UDP socket options
   //extern const UDP_CORK:c_int;
 
 
   /* SOCKET STRUCTURE TYPES */
 
+  extern type sys_in_addr_t;
+  extern type sys_in6_addr_t;
+
   extern type sys_sockaddr_storage_t;
   extern record sys_sockaddr_t {
     var addr:sys_sockaddr_storage_t;
     var len:socklen_t;
+
     proc init() {
       this.complete();
       sys_init_sys_sockaddr_t(this);
     }
+
+    /*
+    Initialize sys_sockaddr_t with provided `family`, `host` and
+    `port`. `host` should be provided in standard notation as per
+    family. Note : `host` isn't resolved using DNS Lookup.
+
+    :arg host: hostname address in ipv4 or ipv6 string notation
+    :type host: `string`
+
+    :arg port: port number
+    :type port: `uint(16)`
+
+    :arg family: domain of socket
+    :type family: `c_int`
+
+    :throws IllegalArgumentError: Upon failure to provide a compatible
+                                  `host` and `family`.
+    */
+    proc initialize(host: c_string, port: uint(16), family: int) throws {
```

Comment:
If you write it as `proc init` then you would run it with e.g. `new sys_sockaddr_t(host, port, family)`. I think `initialize` is probably not a great term since `init` is short for it but is a special function.